billboard.py

```diff
@@ -339,6 +375,51 @@ def getMeta(attribute, ifNoValue=None):
             )
             self.entries.append(entry)
 
+    def _parseYearEndPage(self, soup):
+        # Parse the chart's year from its URL
+        try:
+            href = soup.select_one("link").get("href")
+            self.year = str(re.search(r"/((1|2)\d{3})/", href).group(1))
+        except AttributeError:
+            message = "Could not find a year in the URL."
+            raise BillboardNotFoundException(message)
+
+        # Determine the next and previous year-end chart
+        years = soup.select_one("ul.dropdown__year-select-options")
+        year_links = [li.get("href") for li in years.find_all("a")]
+        idx = [i for i, yl in enumerate(year_links) if self.year in yl]
+        if idx:
+            idx = idx[0]
+            self.previousYear = year_links[min(idx + 1, len(year_links) - 1)]
+            self.nextYear = year_links[max(idx - 1, 0)]
```

Comment:
I think these aren't parsed correctly?

```
>>> import billboard
>>> chart = billboard.ChartData('hot-100-songs', year='2008')
>>> chart.nextYear
'/charts/year-end/2009/hot-100-songs'
>>> chart.previousYear
'/charts/year-end/2007/hot-100-songs'
```

Reply:
It also doesn't work when there is no next year:

```
>>> chart = billboard.ChartData('country-catalog-albums-artists', year='2019')
>>> chart.previousYear
'/charts/year-end/2018/country-catalog-albums-artists'
>>> chart.nextYear
'/charts/year-end/2019/country-catalog-albums-artists'
```

Reply:
Oh yeah, good catch, it should be returning just the year, not the full link. I'll fix that.

How should we handle years that aren't listed in the drop-down menu but may still be valid? I'm inclined to just increment/decrement the year manually, but that approach wouldn't have any knowledge of the page's values.

Reply:
Addressed in [7a56d1a](https://github.com/guoguo12/billboard-charts/pull/77/commits/7a56d1a780a2c358cb06a59fcb280afe6ce89890).

Reply:
I think we shouldn't return years that aren't in the dropdown. That way we can't return the incorrect results you mentioned above.

Reply:
It's unfortunate that the website is inaccurate, but I'd rather return fully correct partial results than run the risk of returning broken results.

Reply:
Sure, that makes sense to avoid potentially returning incorrect results.

Would you also like to prevent the user from manually requesting a year not listed on the drop-down menu?

Reply:
Hmm, I can't make up my mind. On the one hand, it seems pretty aggressive/opinionated, but on the other, I wouldn't want a user to spend hours tracking down the sort of issue you described above.

What are your thoughts?

Reply:
Thanks for the thoughtful consideration!

I'd be inclined to let the user enter whatever year they want, but alert them with some sort of explanatory warning message if they're requesting (or butting up against) a date outside of the chart's drop-down. As a user I would be pretty frustrated if I saw particular years available on the Billboard website (accessed by manually entering their URLs) but I was somehow prevented from accessing them through this package.

Reply:
That seems reasonable. Do you think that should be a [`warnings` warning](https://docs.python.org/3/library/warnings.html)? Or maybe requesting a date outside the dropdown should raise unless you pass `expert=True` or something? (What's conventional in Python?)

Reply:
I think it'd be more natural to use `warnings` to alert the user that they've supplied a maybe "unsupported year" (or whatever term we decide). The user could then choose to silence warnings with `warnings=False` or maybe `warn_on_unsupported_years=False`.

Reply:
Alright, sounds good. 

Python natively supports suppressing warnings, so we probably don't need to include a flag for that.

Reply:
Makes sense! I'll get to work on the updates.

Reply:
Alright, I [added a warning](https://github.com/guoguo12/billboard-charts/blob/41de8e8e55b60fa32aba5c0efdf8ad1b5c41aaf8/billboard.py#L404-L414) that alerts the user if they request a year outside of the year-end chart's drop-down menu.

I also [added a test](https://github.com/guoguo12/billboard-charts/blob/41de8e8e55b60fa32aba5c0efdf8ad1b5c41aaf8/tests/test_year_end_charts.py#L60-L74) for the 1970 Country Hits chart to make sure it raises the warning.

What do you think?

Reply:
Looks good to me.